lib/preprocess/raw.js

```diff
@@ -0,0 +1,27 @@
+const { nodeListToArray } = require('../utils')
+
+// FIXME: rename docxRaw to docxRun or docxRawRun?
+const regexp = /{{docxRaw [^{}]{0,500}}}/
+
+// the problem is that the {{docxRaw}} literal is in a w:t element, which is supposed to only contain literal text,
+// but we want the docxRaw helper to provide a raw XML w:r.
+// Word 365 is not bothered by that, but other docx editors can be.
+// E.g. Word Online displays a broken table and Libreoffice drops the run altogether.
+
+// we find the {{docxRaw}} literal in the w:t element and move it up the tree so it is in its own w:r
+
+module.exports = (files) => {
+  const documentFile = files.find(f => f.path === 'word/document.xml').doc
+  const generalTextElements = nodeListToArray(documentFile.getElementsByTagName('w:t'))
+
+  for (const textEl of generalTextElements) {
+    // there may be more than one docxRaw in a single w:t
+    while (textEl.textContent.includes('{{docxRaw')) {
+      const helperCall = textEl.textContent.match(regexp)[0]
+      const newNode = documentFile.createElement('docxRemove')
+      newNode.textContent = helperCall
+      textEl.parentNode.parentNode.insertBefore(newNode, textEl.parentNode.nextSibling)

```

Comment:
We may want to add an option to specify how many levels we want to go up the tree, e.g. if we want to insert not just a run but a paragraph, a cell, etc.

Reply:
On second thought, I think it would be more convenient to have separate helpers `docxRawRun`, `docxRawParagraph`, `docxRawCell`, etc.

Reply:
> We may want to add an option to specify how many levels we want to go up the tree, e.g. if we want to insert not just a run but a paragraph, a cell, etc.

we like this idea instead of having multiple helpers, you can add a parameter to the helper that symbolize where it is going to be located.

something like this:

```
{{docxRaw xml=xxx replaceParentElement='w:p'}}
```

in this way you can just change the value of the parameter to be `w:c`, `w:r` depending on your needs, this parameter should not have any default, which means that it is required and if the caller does not pass something you should throw error, we will like to have it this way to force the called to think about the shape it is going to create.

Reply:
Done.